File: PP/Lista1/sqrList.ml
(*
    O(n)
    a' list -> a' list
*)
let rec sqrList xs =
    match xs with
    | [] -> []
    | h :: t -> (h*h) :: sqrList t;;

let rec print_list xs =
    match xs with
    | [] -> ()
    | e::l -> print_int e ; print_string " " ; print_list l;;

print_list (sqrList [1;2;3;4;-5]);;

File: PP/Lista1/sqrList.scala
object sqrList {
    def main(args: Array[String]) : Unit = {
        def sqrList(xs : List[Int]) : List[Int] =
            xs match {
                case List() => List()
                case h :: t => (h*h) :: sqrList(t)
            }
        println(sqrList(List(1,2,3,4,5)))
    }
}